src/System.Net.WebSockets/tests/WebSocketTests.netcoreapp.cs

```diff
@@ -66,6 +69,31 @@ public void ValueWebSocketReceiveResult_Ctor_ValidArguments_Roundtrip(int count,
             Assert.Equal(endOfMessage, r.EndOfMessage);
         }
 
+        [OuterLoop] // Connects to external server.
+        [Fact]
+        public async Task WebSocketProtocol_CreateFromConnectedStream_Succeeds()
+        {
+            Socket s = new Socket(SocketType.Stream, ProtocolType.Tcp);
+            await s.ConnectAsync("corefx-net.cloudapp.net", 80);
+            using (NetworkStream ns = new NetworkStream(s))
+            {
+                using (WebSocket socket = WebSocket.CreateFromStream(ns, false, null, TimeSpan.FromSeconds(10)))
```

Comment:
The connection needs to have already upgraded to a websocket connection (meaning it needs to have already gone through the whole sending of the initial headers, gotten back the 101 switch protocols response, etc.).  This hasn't.  That's why the receive fails.

Reply:
This also isn't connecting to an echo server, so even if it did upgrade, the receive would still hang as there's nothing to receive.